raiden/log_config.py

```diff
@@ -1,32 +1,138 @@
+import logging
 import logging.config
-import structlog
 import re
+import sys
 from traceback import TracebackException
 from functools import wraps
-from typing import Dict, Callable, Pattern
+from typing import Dict, Callable, Pattern, Tuple, List
 
+import structlog
 
 DEFAULT_LOG_LEVEL = 'INFO'
+MAX_LOG_FILE_SIZE = 5 * 1024 * 1024
+
+
+def _chain(first_func, *funcs) -> Callable:
+    """Chains a give number of functions.
+    First function receives all args/kwargs. Its result is passed on as an argument
+    to the second one and so on and so forth until all function arguments are used.
+    The last result is then returned.
+    """
+    @wraps(first_func)
+    def wrapper(*args, **kwargs):
+        result = first_func(*args, **kwargs)
+        for func in funcs:
+            result = func(result)
+        return result
+    return wrapper
 
 
-def _get_log_handler(formatter, log_file):
+class LogFilter:
+    """ Utility for filtering log records on module level rules """
+
+    def __init__(self, config: Dict[str, int], default_level: str):
+        """ Initializes a new `LogFilter`
+
+        Args:
+            config: Dictionary mapping module names to logging level
+            default_level: The default logging level
+        """
+        self._should_log = {}
+        # the empty module is not matched, so set it here
+        self._default_level = config.get('', default_level)
+        self._log_rules = [
+            (logger.split('.'), level)
```

Comment:
This may fail when `logger` target is `''`, because it'll end up in the modules-split list being `['']`, which should only be prefix to logger names split list starting with `.`. You should use `logger.split('.') if logger else []`, as `[]` is _list_-prefix of any list.